shared/naturalcrit/codeEditor/fold-code.js

```diff
@@ -0,0 +1,338 @@
+/* eslint-disable max-lines */
+module.exports = {
+	enableCodeFolding : function (CodeMirror) {
```

Comment:
Ok, so all of this file except for `RegisterHomebreweryHelper()` is just copied out from the CodeMirror source, right? Just want to be sure we're looking at the right parts as we try to get the normal `require` to work.

Reply:
Yep! The contents of the `enableCodeFolding` function are from CodeMirror files directly, only the contents of the IIFEs in those files. I made adjustments to them to make them conform to the code style requirements, such as var -> let, const where applicable, and switching `function <name>() {` to `const <name> = function() {` and reordering of the declarations where necessary as a result. There are two comment lines there that explain what code came from what files.

The other function in there is one I made to handle the code folding on pages and only pages. It only matches on `\page` when it is at the beginning of the line, as discussed in the issue.

Reply:
Alright, so I think we definitely should take another look at getting this to work without having to copy and paste all those functions, otherwise this adds a whole layer of extra maintenance that we have to think about if CodeMirror has an update.

@jeddai Do you think you can change this PR to remove all of those functions and get it back as close as possible to how you *think* `require` should work?  Even if it's not functioning at that point it's at least in a state where the rest of us can try our hand debugging it.

Reply:
Yep I'll make that change tonight -- I'm still working on getting it to keep the fold state on change, I have it working but it is a very sketchy solution to say the least. (timeout of 0ms re-folding all the code in the componentDidUpdate function)

Reply:
After adding the new .js files to the project libs code-folding now works with the `require` method!